crates/indexer/src/indexer/fetcher.rs

```diff
@@ -92,29 +92,25 @@ impl Fetcher {
         loop {
             self.ensure_highest_known_version().await;
 
-            let num_missing = self.highest_known_version - self.current_version;
-
-            let num_batches = std::cmp::max(
-                std::cmp::min(
-                    (num_missing as f64 / transaction_fetch_batch_size as f64).ceil() as u64,
-                    self.options.max_tasks as u64,
-                ),
-                1,
-            ) as usize;
-
             info!(
-                num_missing = num_missing,
-                num_batches = num_batches,
                 current_version = self.current_version,
                 highest_known_version = self.highest_known_version,
+                max_batch_size = transaction_fetch_batch_size,
                 "Preparing to fetch transactions"
             );
 
             let fetch_start = chrono::Utc::now().naive_utc();
             let mut tasks = vec![];
-            for i in 0..num_batches {
-                let starting_version =
-                    self.current_version + (i as u64 * transaction_fetch_batch_size as u64);
+            let mut starting_version = self.current_version;
+            let mut num_fetches = 0;
+
+            while num_fetches < self.options.max_tasks
```

Comment:
Is max_tasks === threads?

Reply:
it can be, but I think optimal numbers is max_tasks > threads